packages/consumption/src/modules/requests/itemProcessors/proposeAttribute/ProposeAttributeRequestItemProcessor.ts

```diff
@@ -93,15 +111,60 @@ export class ProposeAttributeRequestItemProcessor extends GenericRequestItemProc
         params: AcceptProposeAttributeRequestItemParametersJSON,
         requestInfo: LocalRequestInfo
-    ): Promise<ProposeAttributeAcceptResponseItem> {
-        const parsedParams: AcceptProposeAttributeRequestItemParameters = AcceptProposeAttributeRequestItemParameters.from(params);
+    ): Promise<ProposeAttributeAcceptResponseItem | AttributeSuccessionAcceptResponseItem> {
+        const parsedParams = AcceptProposeAttributeRequestItemParameters.from(params);
 
         let sharedLocalAttribute: LocalAttribute;
         if (parsedParams.isWithExistingAttribute()) {
-            sharedLocalAttribute = await this.copyExistingAttribute(parsedParams.attributeId, requestInfo);
-        } else {
-            sharedLocalAttribute = await this.createNewAttribute(parsedParams.attribute!, requestInfo);
+            const existingSourceAttribute = await this.consumptionController.attributes.getLocalAttribute(parsedParams.attributeId);
+            if (typeof existingSourceAttribute === "undefined") {
+                throw TransportCoreErrors.general.recordNotFound(LocalAttribute, parsedParams.attributeId.toString());
+            }
+
+            const latestSharedVersion = await this.consumptionController.attributes.getSharedVersionsOfRepositoryAttribute(parsedParams.attributeId, [requestInfo.peer], true);
+
+            if (latestSharedVersion.length === 0) {
+                sharedLocalAttribute = await this.copyExistingAttribute(existingSourceAttribute.id, requestInfo);
+                return ProposeAttributeAcceptResponseItem.from({
+                    result: ResponseItemResult.Accepted,
+                    attributeId: sharedLocalAttribute.id,
+                    attribute: sharedLocalAttribute.content
+                });
+            }
+
+            const latestSharedAttribute = latestSharedVersion[0];
+            if (typeof latestSharedAttribute.shareInfo?.sourceAttribute === "undefined") {
+                throw new Error(`The Attribute ${latestSharedAttribute.id} does not fulfill the requirements of an own shared Attribute.`);
+            }
+
+            if (latestSharedAttribute.shareInfo.sourceAttribute.toString() === existingSourceAttribute.id.toString()) {
+                // return new AttributeAlreadySharedResponseItem
+            }
+
+            const predecessorSourceAttribute = await this.consumptionController.attributes.getLocalAttribute(latestSharedAttribute.shareInfo.sourceAttribute);
+            if (typeof predecessorSourceAttribute === "undefined") {
+                throw TransportCoreErrors.general.recordNotFound(LocalAttribute, latestSharedAttribute.shareInfo.sourceAttribute.toString());
+            }
+
+            if (await this.consumptionController.attributes.isSubsequentInSuccession(predecessorSourceAttribute, existingSourceAttribute)) {
+                let successorSharedAttribute: LocalAttribute;
+                if (existingSourceAttribute.isIdentityAttribute()) {
```

Comment:
Why not use `.isOwnSharedIdentityAttribute()`?

Reply:
Actually, `existingSourceAttribute` must be a RepositoryAttribute, not an own shared IdentityAttribute. But you are right, this can be formulated more precisely.